### PR TITLE
Add realism clean

### PR DESCRIPTION
**Pull request by kfu02**
Cleaned up version of #15 

Linked with: https://github.com/GT-STAR-Lab/het-mamp/pull/37/files

### CODE REVIEW

File: vmas/scenarios/transport.py [hunk 1]
```diff
@@ -22,27 +22,43 @@
...
         self.package_mass = kwargs.get("package_mass", 50)
-        
-        self.add_dense_reward = kwargs.get("add_dense_reward", True)
+        # TODO: implement automated domain randomization here?
+
+        # rewards
         self.agent_package_dist_reward_factor = kwargs.get("agent_package_dist_reward_factor", 0.1)
         self.package_goal_dist_reward_factor = kwargs.get("package_goal_dist_reward_factor", 100)
+        self.interagent_collision_penalty = kwargs.get("interagent_collision_penalty", -1)
+        assert self.interagent_collision_penalty < 0, f"self.interagent_collision_penalty must be < 0, current value is {self.interagent_collision_penalty}!"
```
Comment (kfu02) [inline vmas/scenarios/transport.py]:
```suggestion
        assert self.interagent_collision_penalty <= 0, f"self.interagent_collision_penalty must be <= 0, current value is {self.interagent_collision_penalty}!"
```

File: vmas/scenarios/transport.py [hunk 2]
```diff
@@ -22,27 +22,43 @@
...
         self.capability_mult_min = self.capability_mult_range[0]
         self.capability_mult_max = self.capability_mult_range[1]
         self.capability_representation = kwargs.get("capability_representation", "raw")
 
+        # general world settings
         self.world_semidim = 0.75 
         self.default_agent_radius = 0.03
         self.default_agent_u = 0.6
         self.default_agent_mass = 1.0
+        self.min_collision_distance = 0.05 * self.default_agent_radius # default navigation collision dist is 5% of the agent radius
```
Comment (piercehowell) [inline vmas/scenarios/transport.py:61]: Are agent collisions determined based on distance from center to center of agents, or perimeter to perimeter. If it is the former, then agents would actually visually show collision before a collision is actually registered. This should be checked. If it is the case that it is determined from the center of robots, maybe make it 1.05 agents default radius.

File: vmas/scenarios/transport.py [hunk 3]
```diff
@@ -216,6 +246,23 @@ def reward(self, agent: Agent):
...
+            # penalty (negative rew) for agent-agent collisions
+            for a in self.world.agents:
+                a.agent_collision_rew[:] = 0
+
+            for i, a in enumerate(self.world.agents):
+                for j, b in enumerate(self.world.agents):
+                    if i <= j:
+                        continue
+                    if self.world.collides(a, b):
+                        distance = self.world.get_distance(a, b)
```
Comment (piercehowell) [inline vmas/scenarios/transport.py:258]: See previous comment concerining agents distance for collision calculation

File: vmas/scenarios/transport.py [hunk 4]
```diff
@@ -381,6 +381,59 @@ def default_observation(self, agent: Agent):
...
+            package_obs.append(self.og_package_positions[i])
+            package_obs.append(package.on_goal.unsqueeze(-1))
+            
+            mask = (torch.linalg.vector_norm(package.state.pos - agent.state.pos, dim=-1) < self.package_observation_radius)
+            pkg_state_vec = package.state.pos.clone()
+            pkg_vel_vec = package.state.vel.clone()
+            pkg_dist_to_goal_vec = package.state.pos - package.goal.state.pos
+            agent_dist_to_pkg_vec = package.state.pos - agent.state.pos
+            
+            pkg_state_vec[~mask] = out_of_obs_val
```
Comment (piercehowell) [inline vmas/scenarios/transport.py:407]: Add package rotation to partial observability function observation

File: vmas/scenarios/transport.py [hunk 5]
```diff
@@ -381,6 +381,59 @@ def default_observation(self, agent: Agent):
...
+            ] + capability_repr,
+            dim=-1,
+        )
+
+    def default_observation(self, agent: Agent):
+        # get positions of all entities in this agent's reference frame
+
+        package_obs = []
+        for package in self.packages:
+            package_obs.append(package.state.pos - package.goal.state.pos)
```
Comment (piercehowell) [inline vmas/scenarios/transport.py:437]: Add package rotation state

**Review by piercehowell — CHANGES_REQUESTED** [review]
I've left comments regarding minor changes that should be made. Otherwise, this looks good.

File: vmas/scenarios/transport.py (hunk 2 above)
Comment (kfu02) [reply vmas/scenarios/transport.py:61]: Perimeter to perimeter, see the collision checking code here: https://github.com/GT-STAR-Lab/VectorizedMultiAgentSimulator/blob/main/vmas/simulator/core.py#L1419-L1421

Additionally, I largely ripped the collision distance code from VMAS' navigation environment, where it seems to work for penalizing collisions there.

File: vmas/scenarios/transport.py (hunk 4 above)
Comment (kfu02) [reply vmas/scenarios/transport.py:407]: Done for both, see [commit](https://github.com/GT-STAR-Lab/VectorizedMultiAgentSimulator/pull/18/commits/23d205951a2aed89c7782afbaa5c1a952931a416)

File: vmas/scenarios/transport.py (hunk 3 above)
Comment (kfu02) [reply vmas/scenarios/transport.py:258]: Also perimeter to perimeter (see above)

**Review by piercehowell — APPROVED** [review]

**Pull request merged**